p2p/kademlia/dht.go

```diff
@@ -169,20 +192,43 @@ func (s *DHT) Store(ctx context.Context, data []byte) (string, error) {
 
 // Retrieve data from the networking using key. Key is the base58 encoded
 // identifier of the data.
-func (s *DHT) Retrieve(ctx context.Context, key string) ([]byte, error) {
+func (s *DHT) Retrieve(ctx context.Context, keyType KeyType, key string) ([]byte, error) {
 	decoded := base58.Decode(key)
 	if len(decoded) != B/8 {
 		return nil, fmt.Errorf("invalid key: %v", key)
 	}
 
+	actualKey := s.storedKey(keyType, decoded)
+
+	// retrieve the key/value from local storage
+	value, err := s.store.Retrieve(ctx, actualKey)
+	if err != nil {
+		log.WithContext(ctx).WithError(err).Error("store retrive failed")
```

Comment:
don't wanna return err?

Reply:
@matee81 : in case of error then value is nil, then we will try to request it from peers.